embulk-standards/src/main/java/org/embulk/standards/RenameFilterPlugin.java

```diff
@@ -82,12 +96,381 @@ public PageOutput open(TaskSource taskSource, Schema inputSchema,
         String getRule();
     }
 
+    private interface CharacterTypesRule
+            extends Rule {
+        @Config("pass_types")
+        @ConfigDefault("[]")
+        List<String> getPassTypes();
+
+        @Config("pass_characters")
+        @ConfigDefault("\"\"")
+        String getPassCharacters();
+
+        @Config("replace")
+        @ConfigDefault("\"_\"")
+        @Size(min = 1, max = 1)
+        String getReplace();
+    }
+
+    private interface FirstCharacterTypesRule
+            extends Rule {
+        @Config("replace")
+        @ConfigDefault("null")
+        Optional<String> getReplace();
+
+        @Config("pass_types")
+        @ConfigDefault("[]")
+        List<String> getPassTypes();
+
+        @Config("pass_characters")
+        @ConfigDefault("\"\"")
+        String getPassCharacters();
+
+        @Config("prefix")
+        @ConfigDefault("null")
+        Optional<String> getPrefix();
+    }
+
+    private interface TruncateRule
+            extends Rule {
+        @Config("max_length")
+        @ConfigDefault("128")
+        @Min(0)
+        int getMaxLength();
+    }
+
+    private interface RegexReplaceRule
+            extends Rule {
+        @Config("match")
+        String getMatch();
+
+        @Config("replace")
+        String getReplace();
+    }
+
+    private interface UniqueNumberSuffixRule
+            extends Rule {
+        @Config("delimiter")
+        @ConfigDefault("\"_\"")
+        String getDelimiter();
+
+        @Config("digits")
+        @ConfigDefault("null")
+        Optional<Integer> getDigits();
+
+        @Config("esteem_original_names")
+        @ConfigDefault("true")
+        boolean getEsteemOriginalNames();
```

Comment:
Does this need to be configurable? I think people don't need this option (and omitting this option will documentation easier).